src/app/components/feed/jh-main-feed.directive.js

```diff
@@ -33,6 +35,41 @@
           Janus.attachMediaStream(video, newVal);
         }
       });
+      
+      $timeout(function(){
+        var mainFeed = document.getElementById('main-feed');
+        mainFeed.addEventListener('dblclick', function() {
+          // if video is in fullscreen mode
+          if (
```

Comment:
I don't know right now how to organise the `utilities` functions in Angular 2, but it makes sense, at least for me, to move this code to a separated file/module that you could import and use here. 

Then, you can simplify the directive writing something like 

```javascript
mainFeed.addEventListener('dblclick', function() { toggleFullScreen(mainFeed) })
```

Even more, Iooking for a module example to illustrate my comment I found, accidentally, the [screenfull](https://github.com/sindresorhus/screenfull.js) library, a cross-browser wrapper that you could use if the maintainers allow you to include a small dependency.